tests/unit/s2n_tls13_handshake_state_machine_test.c

```diff
@@ -122,6 +171,13 @@ int main(int argc, char **argv)
     /* Test: A WITH_EARLY_DATA form of every non-full, non-retry handshake exists
      *       and matches the non-WITH_EARLY_DATA form EXCEPT for the END_OF_EARLY_DATA
      *       message and client CCS messages.
+     *= https://tools.ietf.org/rfc/rfc8446#5
+     *= type=test
+     *# An implementation may receive an unencrypted record of type
+     *# change_cipher_spec consisting of the single byte value 0x01 at any
+     *# time after the first ClientHello message has been sent or received
+     *# and before the peer's Finished message has been received and MUST
+     *# simply drop it without further processing.
```

Comment:
How is this relevant to checking for WITH_EARLY_DATA handshakes? The WITH_EARLY_DATA handshake changes when it sends the CCS message, but doesn't change when the CCS message can be received.

Same for most of the tests in this file. They mention CCS messages, but they're really not about when CCS messages can be received. It's really only enforced by the "can receive CCS at any time" tests.